spec/services/match_service_spec.rb

```diff
@@ -67,7 +67,30 @@
       end
     end
 
-    # TODO: matches right teams for !powers of 2
+    [
+      { team_size: 3 },
+      { team_size: 5 },
+      { team_size: 7 },
+      { team_size: 9 },
+      { team_size: 19 },
+      { team_size: 41 },
+      { team_size: 52 },
+      { team_size: 111 }
+
+    ].each do |parameters|
+      it "the right matchups for team numbers that are not powers of 2 (#{parameters[:team_size]})" do
+        team_size = parameters[:team_size]
+        teams = build_list(:team, team_size, tournament: create(:tournament))
+        generated_matches = MatchService.generate_matches teams
+        team_order = []
+        generated_matches.each do |match|
+          match.match_scores.each do |score|
+            team_order << score.team
+          end
+        end
+        expect(team_order).to eq(teams)
```

Comment:
[`match_array`](https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers) is probably better than `eq` here.